test/comprehensive/enc/testdeepcopy.cpp

```diff
@@ -736,6 +736,26 @@ void deepcopy_mix(
     set_sizeparam(s_out_2, 50, 50, 'F');
 }
 
+void deepcopy_nested_countparam_inout(CountParamNestedStruct* s)
+{
+    OE_TEST(s->num == 3);
+    check_countparam(&s->array_of_struct[0], 10, 10, 'A');
+    check_countparam(&s->array_of_struct[1], 20, 20, 'B');
+    check_countparam(&s->array_of_struct[2], 30, 30, 'C');
+
+    set_countparam(&s->array_of_struct[0], 10, 10, 'D', 0);
```

Comment:
What if the callee does `s->num =2;` ? Do we lock that down somewhere?

Reply:
https://github.com/openenclave/oeedger8r-cpp/blob/2b1379d7cff302537bd3f413065c8f9f6c0d602b/test/comprehensive/edl/deepcopy.edl#L53

We previously only support using constant number instead of variable

Reply:
https://github.com/openenclave/oeedger8r-cpp/blob/2b1379d7cff302537bd3f413065c8f9f6c0d602b/test/comprehensive/host/testdeepcopy.cpp#L727

Modifying size/count was locked down here